src/main/java/com/apifest/doclet/Doclet.java

```diff
@@ -82,188 +104,333 @@
-                    if(endpoint != null) {
-                        endpoints.add(endpoint);
+                    ParsedEndpoint parsed = parseEndpoint(doc);
+                    if (parsed != null) {
+                        parsedEndpoints.add(parsed);
                     }
                 }
             }
-
-            generateMappingFile(outputFile);
-        } catch (JAXBException e) {
+            if (docletMode.contains(DocletMode.DOC)) {
+                generateDocsFile(parsedEndpoints, mappingDocsOutputFile);
+            }
+            if (docletMode.contains(DocletMode.MAPPING)) {
+                generateMappingFile(parsedEndpoints, mappingOutputFile);
+            }
+            return true;
+        } catch (ParseException e) {
             System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
             return false;
-        } catch (ParseException e) {
+        } catch (JsonGenerationException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (JsonMappingException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (IOException e) {
+            System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
+            return false;
+        } catch (JAXBException e) {
             System.out.println("ERROR: cannot create mapping file, " + e.getMessage());
             return false;
         }
-        return true;
     }
 
-    private static void generateMappingFile(String outputFile) throws JAXBException {
-        if(outputFile == null || outputFile.length() == 0 || NULL.equals(outputFile)) {
-            outputFile = String.format(DEFAULT_MAPPING_NAME, mappingVersion);
+    private static void validateConfiguration() {
+        String modeInput = System.getProperty("mode");
+        if (modeInput == null) {
+            throw new IllegalArgumentException("mode is invalid.");
         }
-        JAXBContext jaxbContext = JAXBContext.newInstance(Mapping.class);
-        Marshaller marshaller = jaxbContext.createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
-        Mapping mapping = new Mapping();
-        mapping.setVersion(mappingVersion);
-        mapping.setBackend(new Backend(backendHost, backendPort));
-        EndpointsWrapper ends = new EndpointsWrapper();
-        ends.setEndpoints(endpoints);
-        mapping.setEndpointsWrapper(ends);
-        marshaller.marshal(mapping, new File(outputFile));
+        String[] modesSplit = modeInput.split(",");
+        for (String modeSplit : modesSplit) {
+            DocletMode mode = DocletMode.fromString(modeSplit);
+            if (mode == null) {
+                throw new IllegalArgumentException("One of the modes you have specified is invalid: " + modeSplit);
```

Comment:
This exception will be thrown when there is no mode set too and the error message is not appropriate in that case.